Docs: Improve the create pull request guide for contributing (#34799)
grafana/grafana: contribute/create-pull-request.md

- Not contain code that mutates state in reducers or thunks.
- Not contain code that accesses the reducers state slice directly. Instead, the code should use state selectors to access state.

> Pull requests that add/modify unit test (Jest) must:

Reviewer: ```suggestion
Pull requests that add or modify unit tests that are written in Jest must:
```

> Pull requests that add/modify unit test (Jest) must:

Reviewer 2: Should it maybe be clarified that this applies to front end tests specifically?

Author: 🤔 I think because is on the ### Frontend-specific guidelines section does not need to be specific

Reviewer 2: Good point 👍

> Pull requests that add or modify unit tests that are written in Jest must:

Reviewer 3: ```suggestion
Pull requests that add or modify unit tests that are written in Jest must adhere to these guidelines:
```

> - Avoid adding snapshots tests, we are removing current ones incrementally

Reviewer: Why "Avoid adding" and not "Don't add"? Are we looking to completely remove snapshot tests, or are there still valid use cases?

Reviewer 4: We're currently migrating from Enzyme to RTL one test at the time.

Reviewer: I'm not sure I understand how this answers the question? If we don't allow new snapshot tests at all, this should say "Don't add" rather than "Avoid". The latter implies that it's ok to add snapshot tests in some circumstances.

Author: yes, it can be confusing the use of 'avoid adding', I will change it to 'Don't add', because we don't want new snapshots :)

> - Don't add snapshots tests, we are removing current ones incrementally.

Reviewer 3: ```suggestion
- Don't add snapshots tests. We are incrementally removing existing snapshot tests, we don't want more.
```

> - If an existing unit test is written in enzyme, migrate it to RTL (React Testing Library), except  if you’re “just” fixing an issue before a major release (i.e v8), then it’s ok to skip migrating the test to RTL

Reviewer 4: Maybe this is better:
```suggestion
- If an existing unit test is written in enzyme, migrate it to RTL (React Testing Library), except if you’re fixing a bug. Bug fixes usually shouldn't include any bigger refactoring, so it’s ok to skip migrating the test to RTL.
```

> - If an existing unit test is written in enzyme, migrate it to RTL (React Testing Library), except  if you’re “just” fixing an issue before a major release (i.e v8), then it’s ok to skip migrating the test to RTL

Reviewer 2: ```suggestion
- If an existing unit test is written in Enzyme, migrate it to RTL (React Testing Library), except if you’re “just” fixing an issue before a major release (i.e v8), then it’s ok to skip migrating the test to RTL
```

Author: Thank you for the review @Clarity-89 🙌🏾  I made the changes :)

> - If an existing unit test is written in enzyme, migrate it to RTL (React Testing Library), except  if you’re “just” fixing an issue before a major release (i.e v8), then it’s ok to skip migrating the test to RTL

Reviewer: ```suggestion
- If an existing unit test is written in enzyme, then migrate it to RTL (React Testing Library). If you’re “just” fixing an issue before a major release (i.e v8), then it’s ok to skip migrating the test to RTL.
```

Author: @marcusolsson I commit some suggestions, if you can take a second look at this point will be awesome 🙌🏾

> - If an existing unit test is written in Enzyme, migrate it to RTL (React Testing Library), except if you’re fixing a bug. Bug fixes usually shouldn't include any bigger refactoring, so it’s ok to skip migrating the test to RTL.

Reviewer 3: ```suggestion
- If an existing unit test is written in Enzyme, migrate it to RTL (React Testing Library), unless you’re fixing a bug. Bug fixes usually shouldn't include any bigger refactoring, so it’s ok to skip migrating the test to RTL.

```

> - If an existing unit test is written in enzyme, migrate it to RTL (React Testing Library), except if you’re fixing a bug. Bug fixes usually shouldn't include any bigger refactoring, so it’s ok to skip migrating the test to RTL.

Author: ```suggestion
- If an existing unit test is written in Enzyme, migrate it to RTL (React Testing Library), except if you’re fixing a bug. Bug fixes usually shouldn't include any bigger refactoring, so it’s ok to skip migrating the test to RTL.
```
